include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -1,322 +1,118 @@
-    static Cube parse(BitStream &stream, float size, const glm::vec3 &position);
-
-    /// Get the type of the cube.
-    /// @return type of the cube.
-    [[nodiscard]] CubeType type();
-
-    /// Get the number of leaves, this octree contains.
-    /// Leaves are cubes of CubeType::INDENTED or CubeTYPE::FULL.
-    /// @return Number of leaves, this octree contains.
-    [[nodiscard]] std::uint64_t leaves();
-
-    // [[nodiscard]] dynamic_bitset<> bits(); // Bit representation of this cube
-    // [[nodiscard]] vector<array<glm::vec3, 8>> vertices(); // All vertices this cube contains.
-    // [[nodiscard]] vector<array<glm::vec3, 4>> sides(); // All sides this cube contains.
-    /// Get all polygons (triangles) of each cube of this octree.
-    /// @return A vector which contains the three vertices representing a triangle.
-    [[nodiscard]] std::vector<std::array<glm::vec3, 3>> polygons(); // All polygons this cube contains.
-
-    /// Invalidate the cache of this cube / octree (not its children).
-    void invalidate_cache();
-
-    /// Make this octree reactive (update its values when one of its attributes changes).
-    /// @param force Whether to make it reactive again even though the connections were established before.
-    void make_reactive(bool force = false);
+    Cube() = default;
+    explicit Cube(const Type type);
+    Cube(const Type type, const float size, const glm::vec3 &position);
+    Cube(std::weak_ptr<Cube> parent, const Type type, const float size, const glm::vec3 &position);
+    Cube(const Cube &rhs);
+    Cube(Cube &&rhs) noexcept;
+    ~Cube() = default;
+    Cube &operator=(Cube rhs);
+    /// Get child.
+    std::shared_ptr<Cube> operator[](const std::size_t idx);
+    /// Get child.
+    const std::shared_ptr<const Cube> operator[](const std::size_t idx) const;
+
+    static void swap(Cube &lhs, Cube &rhs) noexcept;
+
+    /// Is the current cube root.
+    [[nodiscard]] bool is_root() const noexcept;
+    /// At which child level this cube is.
+    /// root cube = 0
+    [[nodiscard]] std::size_t grid_level() const noexcept;
+    /// Counts the number of Type::SOLID and Type::NORMAL cubes.
+    [[nodiscard]] std::size_t count_geometry_cubes() const noexcept;
+
+    /// Set a new type.
+    void set_type(const Type new_type);
+    /// Get type.
+    [[nodiscard]] Type type() const noexcept;
+
+    /// Get childs.
+    [[nodiscard]] const std::array<std::shared_ptr<Cube>, Cube::SUB_CUBES> &childs() const;
+    /// Get indentations.
+    [[nodiscard]] std::array<Indentation, Cube::EDGES> indentations() const noexcept;
+
+    /// Set an indent by the edge id.
+    void set_indent(const std::uint8_t edge_id, Indentation indentation);
+    /// Indent a specific edge by steps.
+    /// @param positive_direction Indent in  positive axis direction.
+    void indent(const std::uint8_t edge_id, const bool positive_direction, const std::uint8_t steps);
```

Comment:
```suggestion
    void set_indent(std::uint8_t edge_id, Indentation indentation);
    /// Indent a specific edge by steps.
    /// @param positive_direction Indent in  positive axis direction.
    void indent(std::uint8_t edge_id, bool positive_direction, std::uint8_t steps);
```

Reply:
Fixed in d81c9e9576c42a3b00b817bb4727c44dc02ff2d6